src/backsolve_adjoint.jl

```diff
@@ -136,7 +136,13 @@ end
                with a discrete cost function but no specified `dgdu_discrete` or `dgdp_discrete`.
                Please use the higher level `solve` interface or specify these two contributions.")
 
-    @unpack p, u0 = sol.prob
+    p = parameter_values(sol)
+    u0 = state_values(sol, 1)
```

Comment:
This isn't correct. It assumes the first value was saved which isn't always true.

Reply:
Fixed in https://github.com/SciML/SciMLSensitivity.jl/pull/1057/commits/4ef4a9e9a0c69148c6d880290016513bb5418ce2